### ISSUE TITLE
Keyboard Shortcuts Menu Accessibility and I18N Improvements

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#7478]
## Bug Description

The keyboard shortcuts menu should be using the [`<kbd>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/kbd) element. That'd remove the need for all those `aria-label` props due to the element's semantic meaning.

Also, the menu should avoid string concatenation for the actual menu items.

For example, "Hold CMD" would need to be translated to "CMD halten" in German, but that's currently not possible because the shortcut isn't one translatable string:

https://github.com/google/web-stories-wp/blob/06d921a7a7e1080a52cb3ff07c8fe7ad5493809f/assets/src/edit-story/components/keyboardShortcutsMenu/keyboardShortcutList.js#L160-L161

This should be using something like:

```jsx
<TranslateWithMarkup
  mapping={{kbd: <kbd />}}
>
  {
    sprintf(
      /* translators: %s: Cmd/Ctrl key. */
      __('Hold <kbd>%s</kbd>', 'web-stories'),
      cmdOrCtrl
    )
  }
</TranslateWithMarkup>
```

## Screenshots

![Screenshot 2021-05-11 at 10 19 58](https://user-images.githubusercontent.com/841956/117783010-737ba200-b242-11eb-8e12-ae93edb646dc.png)

## Additional Context

- Plugin Version: 1.7.0
- WordPress Version: 5.7
- Operating System: macOS
- Browser: Chrome

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by spacedmonkey** [comment]
Related: #7365

**Comment by swissspidy** [comment]
@BrittanyIRL FYI, assigning this one to pea pod as the original creators of this component (if I'm not mistaken).

**Comment by BrittanyIRL** [comment]
(This was brought into the sprint late in the game because it looked like we would have some extra time. As fate would have it, Karma was going to throw fits and eat up most of that time! this might go back to the backlog. Will check in with Samuel when he's back from PTO next week.)

**Comment by maxyinger** [comment]
for something like the [cmdOrCtrl, `/`] key, how would we like to write the translator comment statement?

**Option 1**
```
    {sprintf(
          /* translators: 1: 'Cmd/Ctrl' key 2: '/' key */
          __('<kbd>%1$s</kbd><kbd>%2$s</kbd>', 'web-stories'),
          cmdOrCtrl,
          '/'
    )}
```

**Option 2**
```
    {sprintf(
          /* translators: 1: Cmd/Ctrl key 2: / key */
          __('<kbd>%1$s</kbd><kbd>%2$s</kbd>', 'web-stories'),
          cmdOrCtrl,
          '/'
    )}
```

**Option 3**
```
    {sprintf(
          /* translators: %s: Cmd/Ctrl key */
          __('<kbd>%s</kbd><kbd>/</kbd>', 'web-stories'),
          cmdOrCtrl,
    )}
```

**Comment by swissspidy** [comment]
I don't think we need to translate simple shortcuts that don't include any text, such as `[cmdOrCtrl, '/']`. Just the ones where we have "partial" text like "or double-click", "or", and "Hold".

If anything, I'd go with option 3. Because option 1 and 2 don't include anything translatable at all. (How do you translate `%1$s %2$s` to Spanish?)

Taking `Shift, Up, 'or', Down` as an example, we could write that like so:

```js
'<kbd>' +
printf(
  /* translators: 1: Shift key. 2: Up key. 3: Down key. */
  __('<kbd>%1$s</kbd><kbd>%2$s</kbd> or <kbd>%3$s</kbd>', 'web-stories'),
  SPECIAL_KEYS.SHIFT,
  SPECIAL_KEYS.UP,
  SPECIAL_KEYS.DOWN
)
+ '</kbd>'
```

(Moving the wrapping outer `<kbd>` out of the translatable string makes this easier for translators)

Which would result in final output 

```html
<kbd><kbd>⇧</kbd><kbd>(Up Icon)</kbd>or<kbd>(Down Icon)</kbd></kbd>
```

**Comment by maxyinger** [comment]
Perfect. Sounds good. Thanks for the guidance!

**Comment by maxyinger** [comment]
one other quick question. Our shift constant looks like this:
```
  SHIFT: {
    symbol: '⇧',
    title: _x('Shift', 'The keyboard key "Shift"', 'web-stories'),
  },
```

Should the final rendered result just look like this?
```
<kbd>{SPECIAL_KEYS.SHIFT.symbol}</kbd>
```

Should the title property be included anywhere in there? or is `<kbd>⇧</kbd>` already language agnostic? Not familiar with keyboards in other languages and how that all works out, so still trying to learn a bit here.

Is the `SPECIAL_KEYS[someKey].title` property needed at all once we switch to rendering with `<kbd>`?

**Comment by swissspidy** [comment]
Let's put the `title` in an `aria-label` for these special keys.

```jsx
<kbd aria-label="Shift">⇧</kbd>
```

If that doesn't work well with screen readers, we might need to do something like this:

```jsx
<kbd aria-hidden="true">⇧</kbd><VisuallyHidden>Shift</VisuallyHidden>
```

**Comment by maxyinger** [comment]
Awesome. Will go with `aria-label` for now and try it out. Thanks!

**Comment by maxyinger** [comment]
@swissspidy Sorry for this being the ticket with never ending questions. 

For a key combination with just a special key and an alphabetical key, is there any need to translate?
```
        shortcut: (
            <kbd>
              <TranslateWithMarkup mapping={{ kbd: <kbd />, cmdOrCtrlSymbol: cmdOrCtrl.symbol }}>
                {sprintf(
                  /* translators: %s: Cmd/Ctrl key.*/
                  __(
                    '<kbd aria-label="%s"><cmdOrCtrlSymbol /></kbd><kbd>K</kbd>',
                    'web-stories'
                  ),
                  cmdOrCtrl.title
                )}
              </TranslateWithMarkup>
            </kbd>
        ),
```
vs
```
        shortcut: (
            <kbd>
              <kbd aria-label={cmdOrCtrl.title}>{cmdOrCtrl.symbol}</kbd>
              <kbd>{'K'}</kbd>
            </kbd>
        ),
```

**Comment by swissspidy** [comment]
> For a key combination with just a special key and an alphabetical key, is there any need to translate?

I don‘t think so, no. Let‘s keep those cases simple.

**Comment by BrittanyIRL** [comment]
@csossi this just needs regression testing for the shortcut menu. the translations were updated.

**Comment by csossi** [comment]
Verified in QA
